src/utils.h

```diff
@@ -215,8 +215,12 @@ static inline u8 mask8(u64 addr, u8 clear, u8 set)
     })
 #define mrs(reg) _mrs(reg)
 
-#define _msr(reg, val) ({ __asm__ volatile("msr\t" #reg ", %0" : : "r"(val)); })
-#define msr(reg, val)  _msr(reg, val)
+#define _msr(reg, val)                                                                             \
+    ({                                                                                             \
+        u64 __val = (u64)val;                                                                      \
```

Comment:
Double underscore prefix is reserved to be only used by C compiler and library peoples. ;) C standard permits arbitrary breakage of stuff with double underscore prefixes, so don't use it.

Reply:
I know all identifiers begin with either two underscores or an underscore followed by a capital letter are reserved names.
But this is a general practice when used by macros inside a block scope  '{' and '}' . This approach is widely used by Linux kernel, some examples :
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/arm64/include/asm/sysreg.h#n1076

Reply:
Fair enough.